### PR TITLE
[css-color-5] Adjustments to edge cases and set notation for hue interpolation types

### PR DESCRIPTION
**Pull request by dbaron**
This makes a few adjustments to the definitions of hue interpolation
types to follow up on the discussions in #4735.

First, it adjusts the pseudo-code so that for 'shorter' and 'longer',
angles that differ by 180 degrees will always go in the decreasing
direction, as described in
https://github.com/w3c/csswg-drafts/issues/4735#issuecomment-650254063

Second, it adjusts the set notation to match the pseudo-code.  (They
were not matching even prior to this change.)  The set notation
previously seemed to assume that there were absolute-value functions
present that were not actually there.  However, given the asymmetry of
always going in the decreasing direction, it's more correct to write it
without absolute values.

### CODE REVIEW

**Comment by dbaron** [comment]
(This is following up to the changes made in 4e88f2c4ac6983857fce23599f2f42d03d3d81c1.)

File: css-color-5/Overview.bs [hunk 1]
```diff
@@ -745,20 +745,20 @@ Unless the type of hue interpolation is ''specified'', both angles need to be co
...
 	}
-	else if (θ₂ - θ₁ <= -180) {
+	else if (θ₂ - θ₁ < -180) {
 		θ₂ += 360;
 	}
 	</pre>
 
 : ''longer''
-:: Angles are adjusted so that θ₂ - θ₁ ∈ [180, 360). In pseudo-Javascript:
+:: Angles are adjusted so that θ₂ - θ₁ ∈ {(-360, -180], 0, (180, 360)}. In pseudo-Javascript:
```
Comment (LeaVerou) [inline css-color-5/Overview.bs:759]: I find this rather hard to read, can we prioritize readability over conciseness?

File: css-color-5/Overview.bs [hunk 2]
```diff
@@ -745,20 +745,20 @@ Unless the type of hue interpolation is ''specified'', both angles need to be co
...
 	<pre>
-	if (0 < θ₂ - θ₁ < 180) {
+	if (0 < θ₂ - θ₁ <= 180) {
```
Comment (LeaVerou) [inline css-color-5/Overview.bs:761]: Why is this needed? Currently on the equal case, you just don't add 360. It's still well-defined. Is there a reason why we want to enter this clause?

Comment (dbaron) [reply css-color-5/Overview.bs:761]: The reason for wanting to do this is to follow the principle from @svgeesus that angles that differ by 180 should go in the `decreasing` direction.  Without this change, the direction they go would depend on what angles they were, since it would be determined by the constraining to `[0, 360)`.

File: css-color-5/Overview.bs (hunk 1 above)
Comment (dbaron) [reply css-color-5/Overview.bs:759]: By just changing it to an English sentence rather than set notation?  Or some other way?

File: css-color-5/Overview.bs (hunk 2 above)
Comment (LeaVerou) [reply css-color-5/Overview.bs:761]: After thinking about it some more, I think if θ₂ - θ₁ == 180 (so θ₂ == θ₁ + 180) then interpolation should be the same as `increasing`, whereas when θ₂ - θ₁ == -180 (so θ₂ == θ₁ - 180), interpolation should be the same as decreasing. This seems to be less surprise-inducing than going in the same direction regardless of whether it's θ₂ < θ₁ or θ₂ > θ₁.
I'll think of how to put that in these algorithms tomorrow (or if you want to give it a shot, fine by me!)

File: css-color-5/Overview.bs (hunk 1 above)
Comment (LeaVerou) [reply css-color-5/Overview.bs:759]: 0 is not needed if we make the sets closed, and we can combine [-360, -180] and [180, 360] by using the absolute value of the difference.

File: css-color-5/Overview.bs (hunk 2 above)
Comment (svgeesus) [reply css-color-5/Overview.bs:761]: Yes, @LeaVerou convinced me that basing the direction at exactly 180 on the sign of the angle difference made more sense.

File: css-color-5/Overview.bs (hunk 1 above)
Comment (dbaron) [reply css-color-5/Overview.bs:759]: I'm still pretty hesitant to write a rule that says that interpolation of *identical* hues goes around the circle -- both because we'd have to choose which way, and because I think it might be problematic if we applied such rules to transitions and animations, which I think may depend in some ways on interpolation of identical values not going through other values.

I'll make the other changes shortly.

Comment (LeaVerou) [reply css-color-5/Overview.bs]: Makes sense, yeah.

**Comment by LeaVerou** [comment]
LGTM, and even if we tweak further eventually, the changes bring it to a much better state than the current one, so merging.
